File: backend/app/models/email.py
from datetime import datetime
from enum import Enum
from typing import List, Optional, Dict, Any
from pydantic import BaseModel, Field


class SensitivityLevel(str, Enum):
    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"
    CRITICAL = "critical"


class Department(str, Enum):
    GENERAL = "general"
    ENGINEERING = "engineering"
    PRODUCT = "product"
    MARKETING = "marketing"
    SALES = "sales"
    FINANCE = "finance"
    HR = "hr"
    LEGAL = "legal"
    OTHER = "other"


class PIIType(str, Enum):
    NONE = "none"
    NAME = "name"
    EMAIL = "email"
    PHONE = "phone"
    ADDRESS = "address"
    SSN = "ssn"
    PASSPORT = "passport"
    CREDIT_CARD = "credit_card"
    BANK_ACCOUNT = "bank_account"
    DOB = "date_of_birth"
    SALARY = "salary"
    OTHER = "other"


class ReviewStatus(str, Enum):
    PENDING = "pending"
    APPROVED = "approved"
    REJECTED = "rejected"


class EmailPreview(BaseModel):
    id: str
    subject: str
    sender: str
    received_date: datetime
    snippet: str


class EmailFilter(BaseModel):
    folder_id: Optional[str] = None
    start_date: Optional[datetime] = None
    end_date: Optional[datetime] = None
    keywords: Optional[List[str]] = None
    sender: Optional[str] = None


class EmailAttachment(BaseModel):
    id: str
    name: str
    content_type: str
    size: int
    content: Optional[str] = None


class EmailContent(BaseModel):
    id: str
    internet_message_id: str
    subject: str
    sender: str
    sender_email: str
    recipients: List[str]
    cc_recipients: Optional[List[str]] = None
    received_date: datetime
    body: str
    is_html: bool = False
    folder_id: str
    folder_name: str
    attachments: Optional[List[EmailAttachment]] = None
    importance: str = "normal"


class EmailAnalysis(BaseModel):
    sensitivity: SensitivityLevel = Field(..., description="How sensitive is this information")
    department: Department = Field(..., description="Which department this knowledge belongs to")
    tags: List[str] = Field(..., description="Relevant tags for categorizing this content")
    is_private: bool = Field(..., description="Whether this contains private/confidential information")
    pii_detected: List[PIIType] = Field(default_factory=list, description="Types of personal identifiable information detected")
    recommended_action: str = Field(..., description="Recommended action (store/exclude)")
    summary: str = Field(..., description="Brief summary of the content")
    key_points: List[str] = Field(..., description="Key knowledge points extracted")


class EmailReview(BaseModel):
    email_id: str
    content: EmailContent
    analysis: EmailAnalysis
    status: ReviewStatus = ReviewStatus.PENDING
    reviewed_at: Optional[datetime] = None
    reviewer_id: Optional[str] = None
    review_notes: Optional[str] = None


class EmailApproval(BaseModel):
    approved: bool
    notes: Optional[str] = None


class EmailVectorData(BaseModel):
    id: str
    email_id: str
    content: str
    metadata: Dict[str, Any]
    embedding: Optional[List[float]] = None
    created_at: datetime = Field(default_factory=datetime.now)

File: frontend/src/types/email.ts
export enum SensitivityLevel {
  LOW = "low",
  MEDIUM = "medium",
  HIGH = "high",
  CRITICAL = "critical"
}

export enum Department {
  GENERAL = "general",
  ENGINEERING = "engineering",
  PRODUCT = "product",
  MARKETING = "marketing",
  SALES = "sales",
  FINANCE = "finance",
  HR = "hr",
  LEGAL = "legal",
  OTHER = "other"
}

export enum PIIType {
  NONE = "none",
  NAME = "name",
  EMAIL = "email",
  PHONE = "phone",
  ADDRESS = "address",
  SSN = "ssn",
  PASSPORT = "passport",
  CREDIT_CARD = "credit_card",
  BANK_ACCOUNT = "bank_account",
  DOB = "date_of_birth",
  SALARY = "salary",
  OTHER = "other"
}

export enum ReviewStatus {
  PENDING = "pending",
  APPROVED = "approved",
  REJECTED = "rejected"
}

export interface EmailPreview {
  id: string;
  subject: string;
  sender: string;
  received_date: string;
  snippet: string;
}

export interface EmailFilter {
  folder_id?: string;
  start_date?: string;
  end_date?: string;
  keywords?: string[];
  sender?: string;
}

export interface EmailAttachment {
  id: string;
  name: string;
  content_type: string;
  size: number;
  content?: string;
}

export interface EmailContent {
  id: string;
  internet_message_id: string;
  subject: string;
  sender: string;
  sender_email: string;
  recipients: string[];
  cc_recipients?: string[];
  received_date: string;
  body: string;
  is_html: boolean;
  folder_id: string;
  folder_name: string;
  attachments?: EmailAttachment[];
  importance: string;
}

export interface EmailAnalysis {
  sensitivity: SensitivityLevel;
  department: Department;
  tags: string[];
  is_private: boolean;
  pii_detected: PIIType[];
  recommended_action: string;
  summary: string;
  key_points: string[];
}

export interface EmailReviewItem {
  email_id: string;
  content: EmailContent;
  analysis: EmailAnalysis;
  status: ReviewStatus;
  reviewed_at?: string;
  reviewer_id?: string;
  review_notes?: string;
}

export interface EmailApproval {
  approved: boolean;
  notes?: string;
}
